### ISSUE TITLE
socket-send returns NIL on ABCL

### ISSUE THREAD

**Issue by Uthar** [usocket/usocket#82]
**Expected behavior**: 
socket-send returns the number of bytes written.

**Actual behavior**: 
Returns NIL, which causes a TYPE-ERROR to be signaled from the UDP socket-server before crashing.

**Steps to replicate**:
```
(usocket:socket-server "127.0.0.1" 4440 #'print nil :protocol :datagram :in-new-thread t)
(setf sock (usocket:socket-connect "127.0.0.1" 4440 :protocol :datagram))
(usocket:socket-send sock (make-array 65 :initial-element 0) 65)
```

**Version found in**:
b0f5f0fa554499df9190c1af850a180ea95c008

**Patch**:
```patch
--- backend/abcl.lisp
+++ backend/abcl.lisp
@@ -351,7 +351,8 @@
     (loop for i from offset below (+ size offset)
        do (setf (jarray-ref byte-array i) (*->byte (aref buffer i))))
     (with-mapped-conditions (usocket host)
-      (jcall $@send/1 socket packet))))
+      (jcall $@send/1 socket packet)
+      size)))
 
 ;;; TODO: return-host and return-port cannot be get ...
 (defmethod socket-receive ((usocket datagram-usocket) buffer length

```

**Comment by binghe** [comment]
Hi, thanks for the detailed issue reports. Fixed in 3f3f5b7a using your patch.

**Comment by Uthar** [comment]
Thank you for fixing.
